js/private/js_run_devserver.bzl

```diff
@@ -113,8 +116,19 @@ def _impl(ctx):
     if ctx.attr.tool and ctx.attr.command:
         fail("Only one of tool or command may be specified")
 
+    transitive_npm_links = []
+
+    # The .to_list() calls here are intentional and cannot be avoided; they should be small sets of
+    # files as they only include direct npm links (node_modules/foo) and the virtual store tree
+    # artifacts those symlinks point to (node_modules/.aspect_rules_js/foo@1.2.3/node_modules/foo)
+    for i in _gather_npm_linked_packages(srcs = [], deps = ctx.attr.data).transitive.to_list():
+        for f in i.files.to_list():
```

Comment:
Could we combine the 2 loops to at least only call `to_list()` once? Or would it be twice? 🤔

Reply:
Twice is needed. Once to get the list of NpmLinkedPackageInfo providers and then a 2nd time to get the list of files in each one

Reply:
That's 1 + N, I was wondering about 2...

```
transitive = _gather_npm_linked_packages(srcs = [], deps = ctx.attr.data).transitive.to_list()
files = depset(transitive = [f for f in transitive.to_list()]).to_list()
```
... or something like that? 🤷

Reply:
Oh I see what you mean. Yeah, this can be improved as you suggest with:
```
    # The .to_list() calls here are intentional and cannot be avoided; they should be small sets of
    # files as they only include direct npm links (node_modules/foo) and the virtual store tree
    # artifacts those symlinks point to (node_modules/.aspect_rules_js/foo@1.2.3/node_modules/foo)
    transitive_npm_links_files = []
    transitive_npm_links = depset(transitive = [i.files for i in _gather_npm_linked_packages(srcs = [], deps = ctx.attr.data).transitive.to_list()])
    for f in transitive_npm_links.to_list():
            # don't include the virtual store tree artifact; only the node_module link is needed
            if not "/.aspect_rules_js/" in f.path:
                transitive_npm_links_files.append(f)
```